frontend/src/metabase/containers/CreateCollectionOnTheGo.tsx

```diff
@@ -0,0 +1,66 @@
+import { useState, createContext, useCallback, ReactElement } from "react";
+import { t } from "ttag";
+import { Collection, CollectionId } from "metabase-types/api";
+import CreateCollectionModal from "metabase/collections/containers/CreateCollectionModal";
+import { NewCollectionButton } from "./CreateCollectionOnTheGo.styled";
+
+interface State {
+  enabled: boolean;
+  resumedValues: any;
+  openCollectionId?: CollectionId;
+}
+
+interface Props {
+  children: (resumedValues: any) => ReactElement;
+}
+
+interface ButtonProps {
+  resumedValues: any;
+  openCollectionId?: CollectionId;
+}
+
+type ButtonType = (buttonProps: ButtonProps) => ReactElement;
+export const CreateCollectionOnTheGoButtonContext = createContext<ButtonType>(
```

Comment:
Is using context provides enough benefits here? It looks like the props drilling would not be awful in this case. But with this approach `FormCollectionPicker` gets an implicit dependency from `CreateCollectionOnTheGo`. Since there is no deep nesting, I'd lean more towards just having a prop like `onCreateNewCollection`. Wdyt?

I also noticed that the functionality does not work correctly when you move a collection to a another collection you create right from the modal

Reply:
Yes I agree now. 151e3f1cd60afc855737dd3755c0554b04a08dd8

I was able to move a collection to a collection created on the go.

Reply:
@shaunlebron I ran into https://github.com/metabase/metabase/issues/33026